homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -35,7 +36,10 @@ async def _async_update_data(self) -> dict[str, MowerAttributes]:
             await self.api.connect()
```

Comment:
Side note: The type annotation for `api` is incorrect. It should be `AutomowerSession` instead of `AsyncConfigEntryAuth`.

Reply:
Side note: In the library replace `loop.create_task` and `asyncio.ensure_future` with `asyncio.create_task`. The other ones are legacy.